ext/openssl/ossl_x509cert.c

```diff
@@ -707,6 +707,36 @@ ossl_x509_eq(VALUE self, VALUE other)
     return !X509_cmp(a, b) ? Qtrue : Qfalse;
 }
 
+#ifdef HAVE_I2D_RE_X509_TBS
+/*
+ * call-seq:
+ *    cert.tbs_bytes => string
+ * 
+ * Returns the DER-encoded bytes of the certificate's to be signed certificate.
+ * This is mainly useful for validating embedded certificate transparency signatures.
+ */
+static VALUE
+ossl_x509_tbs_bytes(VALUE self)
+{
+    X509 *x509;
+    int len;
+    unsigned char *p0, *p1;
+    VALUE str;
+
+    GetX509(self, x509);
+    len = i2d_re_X509_tbs(x509, NULL);
+    if (len <= 0) {
+    ossl_raise(eX509CertError, NULL);
+    }
+    str = rb_str_new(NULL, len);
+    p0 = p1 = (unsigned char *)RSTRING_PTR(str);
+    i2d_re_X509_tbs(x509, &p0);
+    assert(p0 - p1 == len);
```

Comment:
The return value of `i2d_re_X509_tbs()` should be checked.

Also, please use `ossl_str_adjust` instead `assert`.

Reply:
Sorry, I was copying from `ossl_asn1prim_to_der`. I'm happy to accept any suggestions you make here, but specific diffs you have in mind would be helpful since I was already copying from existing code in the gem

Reply:
Please see #755. The usage in `ossl_asn1prim_to_der()` was problematic. All `i2d_*()` functions may return a negative number on error and we want to check it.